### DISCUSSION TITLE
High Availability Embedded etcd is not LEARNER

### DISCUSSION THREAD

**Discussion by RhysZhai** [k3s-io/k3s#7979] [Q&A] [answered]
**Environmental Info:**
K3s Version:  1.24.17

Node(s) CPU architecture, OS, and Version: 

Cluster Configuration:

**Describe the bug:**
[root@node0 ~]# etcdctl member list   --cacert=/data/k3s/server/tls/etcd/server-ca.crt   --cert=/data/k3s/server/tls/etcd/client.crt   --key=/data/k3s/server/tls/etcd/client.key --write-out="table"
+------------------+---------+----------------+-----------------------------+-----------------------------+------------+
|        ID        | STATUS  |      NAME      |         PEER ADDRS          |        CLIENT ADDRS         | IS LEARNER |
+------------------+---------+----------------+-----------------------------+-----------------------------+------------+
| 58aab802ac34a961 | started | node2-d5aa31d8 | https://10.106.108.219:2380 | https://10.106.108.219:2379 |      false |
| 5ded2084fe0bd6eb | started | node0-6b5275f6 | https://10.106.108.217:2380 | https://10.106.108.217:2379 |      false |
| 87b895f95fdc9c1d | started | node1-20ea1b9d | https://10.106.108.218:2380 | https://10.106.108.218:2379 |      false |

is not LEARNER

**Steps To Reproduce:**

- Installed K3s: 
- Installed etcdctl
`ETCD_VERSION="v3.5.5"
ETCD_URL="https://github.com/etcd-io/etcd/releases/download/${ETCD_VERSION}/etcd-${ETCD_VERSION}-linux-amd64.tar.gz"
curl -sL ${ETCD_URL} | sudo tar -zxv --strip-components=1 -C /usr/local/bin`
etcdctl member list   --cacert=/data/k3s/server/tls/etcd/server-ca.crt   --cert=/data/k3s/server/tls/etcd/client.crt   --key=/data/k3s/server/tls/etcd/client.key --write-out="table" 
is not LEARNER 
**Expected behavior:**

**Actual behavior:**

**Additional context / logs:**

**Comment by dereknola** [comment, marked as answer]
Once servers successfully join the cluster, they are no longer learners, but full members. There isn't a problem with the setup you are describing. See https://etcd.io/docs/v3.3/learning/learner/#raft-learner

**Reply by brandond** [reply]
You can also see it joining as learner and being promoted to full voting member, if you check the k3s logs.
